plugins/modules/gitlab_group_access_token.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Zoran Krleza (zoran.krleza@true-north.hr)
+# Based on code:
+# Copyright (c) 2019, Guillaume Martinez (lunik@tiwabbit.fr)
+# Copyright (c) 2018, Marcus Watkins <marwatk@marcuswatkins.net>
+# Copyright (c) 2013, Phillip Gentry <phillip@cx.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+module: gitlab_group_access_token
+short_description: Manages GitLab group access tokens
+description:
+  - Creates, gets and revokes group access tokens.
+author:
+  - Zoran Krleza (@pixslx)
+requirements:
+  - python-gitlab >= 3.1.0
+extends_documentation_fragment:
+  - community.general.auth_basic
+  - community.general.gitlab
+  - community.general.attributes
+notes:
+  - Access tokens can not be changed. If a parameter needs to be changed, acceess token has to be recreated.
+  - Token string is contained in the result only when access token is created or recreated. It can not be fetched afterwards.
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+
+options:
+  group:
+    description:
+      - ID or full path of group in the form of group/subgroup.
+    required: true
+    type: str
+  name:
+    description:
+      - Access token's name.
+    required: true
+    type: str
+  scopes:
+    description:
+      - Scope of the access token.
+    required: true
+    type: list
+    elements: str
+    aliases: ["scope"]
+    choices: ["api", "read_api", "read_registry", "write_registry", "read_repository", "write_repository", "create_runner", "ai_features", "k8s_proxy"]
+  access_level:
+    description:
+      - Access level of the access token.
+    type: str
+    default: maintainer
+    choices: ["guest", "reporter", "developer", "maintainer", "owner"]
+  expires_at:
+    description:
+      - Expiration date of the access token in YYYY-MM-DD format.
+    type: str
+    required: true
+  recreate:
```

Comment:
I think it's better to name this `allow_recreate`. I would expect `recreate: true` to **always** recreate the token.
```suggestion
  allow_recreate:
```
Alternatively, convert this option to a string, with choices `never`, `always`, `state_change` (or something like that). Then users can say exactly when they want a token to be recreated.

Reply:
I agree. Recreate option changed to str with suggested choices.